samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -64,21 +65,25 @@
   private String newJobModelVersion;  // version published in ZK (by the leader)
   private JobModel jobModel;
 
-  public ZkJobCoordinator(String processorId, String groupId, Config config, ScheduleAfterDebounceTime debounceTimer, ZkUtils zkUtils,
+  public ZkJobCoordinator(String groupId, Config config, ScheduleAfterDebounceTime debounceTimer,
```

Comment:
Why is processorId removed from here?

The ZkJobCoordinator is passing the first constructor argument as processorId and you are mapping it to groupId here. Please correct it. 

Also, please clarify what groupId means through documentation